src/middleware/auth-modal-middleware.js

```diff
@@ -0,0 +1,23 @@
+import { REGISTER, LOGIN } from '../store/actions/action-types';
+import ShowModal from '../store/actions/change-form-action';
+
+export default function showModalMiddleware({
+  dispatch,
+}) {
+  return next => (action) => {
+    if (action.type === REGISTER) {
+      dispatch(ShowModal({
+        isRegister: true,
+        modalShow: true,
+      }));
+    }
+    if (action.type === LOGIN) {
+      dispatch(ShowModal({
+        isRegister: false,
+        modalShow: true,
+      }));
+    }
+
+    return next(action);
+  };
+}
```

Comment:
This file contains actions, kindly change place it in the actions directory and rename it to something line AuthActions.js